Uswds-navbar-v2

ojbravo opened the pull request:
## Changes

Added USWDS Navbar w/ Algolia Search

## Testing

Increased added 5s timeout to Filing.spec test to ensure it passes.

meissadia reviewed (changes requested):
- User name + logout wrap awkwardly on narrow screens

  <img width="857" alt="Screen Shot 2023-05-22 at 5 03 01 PM" src="https://github.com/cfpb/hmda-frontend/assets/2592907/cce1da72-d5fa-4ad6-a0eb-ec66647c253e">

ojbravo commented on cypress/e2e/filing/Filing.spec.js:
```diff
@@ -145,13 +145,14 @@ describe("Filing", function() {
             // Action: Review Quality Edits
             cy.get('.NavButtonContainer > .NavButton').click()
-            cy.wait(ACTION_DELAY)
+            cy.wait(5000)
```
Update made.

ojbravo commented:
Corrected logout location via css update, added NAV_DELAY test variable, and removed commented code. Will work on simplifying <ShowUserName /> component.

ojbravo commented on src/common/ShowUserName.jsx:
```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from 'react'
+import { logout } from '../filing/utils/keycloak.js'
+import { getKeycloak } from './api/Keycloak.js'
+
+export const ShowUserName = (props) => {
+  let userName
+  const [isLoggedIn, setIsLoggedIn] = useState(props.isLoggedIn)
+  isLoggedIn ? (userName = getKeycloak().tokenParsed.name) : null
+  
+  useEffect(() => {
+    setIsLoggedIn(props.isLoggedIn);
+  }, [props.isLoggedIn]);
+
+  const handleLogout = e => {
+    setIsLoggedIn(false)
+    userName = ''
+    e.preventDefault()
+    logout()
+  }
+  
+  return (
+    <>
+      {isLoggedIn ? (
+        <div className='user'>{userName} <button onClick={handleLogout}>Logout</button></div>
+      ) : null}
+    </>
+  )
+}
```
I'll look into this one... Good looking out!

meissadia commented on src/common/Header.jsx:
```diff
@@ -15,49 +25,99 @@ export const hideHeaderFooter = (path) => {
+                            ? 'usa-nav__link usa-current'
+                            : 'usa-nav__link'
+                        }
+                        //target={link.name === 'Filing' ? '_blank' : null}
```
```suggestion
```

Michaeldremy reviewed (approved):
Look good to me!

meissadia commented on src/common/Header.jsx:
```diff
@@ -15,49 +25,99 @@ export const hideHeaderFooter = (path) => {
+                            : 'usa-nav__link'
+                        }
+                        //target={link.name === 'Filing' ? '_blank' : null}
+                        rel={
+                          link.name === 'Filing' ? 'noopener noreferrer' : null
+                        }
+                      >
+                        {link.name}
+                      </Link> :
+                      <>
+                        <button
+                          type="button"
+                          className="usa-accordion__button usa-nav__link"
+                          aria-expanded="false"
+                          aria-controls={link.name}
+                        >
+                          <span>{link.name}</span>
+                        </button>
+                        <ul id={link.name} className="usa-nav__submenu">
+                          {link.submenu.map(sublink => (<li key={sublink.name} className="usa-nav__submenu-item">
+                            {sublink.name == "Developer APIs" || sublink.name == "FAQs" ?
+                              <Link
+                                to={sublink.href}
+                                onClick={() => forceReload(sublink.href)}
+                              >
+                                {sublink.name}
+                              </Link> :
+                              !sublink.href ? 
+                                <div className="subMenuHeading">{sublink.name}</div> :
+                                <Link
+                                  to={sublink.href}
+                                >
+                                  {sublink.name}
+                                </Link>
+                            }
+                          </li>)
+                          )}
+                        </ul>
+                      </>
+                    }
```
This feels a bit dense with all the ternary operators.  Consider breaking things out into a few sub-components.

Michaeldremy replied:
Going to leave this as is for now. Can come back and refactor later.

meissadia commented on src/common/Header.jsx:
```diff
@@ -15,49 +25,99 @@ export const hideHeaderFooter = (path) => {
+            </button>
+            <ul className="usa-nav__primary usa-accordion">
               {links.map(link => {
                 let isActive = pathname.match(new RegExp('^' + link.href))
                 if(link.href === '/') isActive = link.href === pathname
```
This doesn't seem to work for the `Filing` page
